### PR TITLE
Specific Item selectability

### PR DESCRIPTION
**Pull request by jczacharia**
This PR adds the feature: the ability to make specific items selectable.

Demo: http://plnkr.co/edit/qJSULoHlFF7GvcoWhhsK

### CODE REVIEW

**Comment by jczacharia** [comment]
I found a bug where shift+click will still select all items in a group regardless of the selectability flag. Will try to fix this.

**Comment by yotamberk** [comment]
Hey @jczacharia , It seems you added the wrong example. I don't see this feature in the current example.

**Comment by jczacharia** [comment]
@yotamberk  Are you looking at the example in the PR or the demo?

**Comment by yotamberk** [comment]
Demo: http://plnkr.co/edit/qJSULoHlFF7GvcoWhhsK

**Comment by jczacharia** [comment]
In the demo, set `item selectability of 'no subgroup'` to true then select the bottom left item that says `no subgroup`. Notice how it is selectable.

Then set `item selectability of 'no subgroup'` to false, notice how you cannot select it.

This PR allows items to be not selectable even if `selectable: true` in the _timeline's_ config options.

I apologize if the demos weren't the most straightforward.

**Comment by yotamberk** [comment]
I don;t understand still why would we need this option. Isn't it enough to simply set the item you don;t want to be selectable as `selectable: false`?

**Comment by jczacharia** [comment]
As of now, there is no `selectable` option for items. Only for the timeline which makes either all the items selectable or not. I have a use case where I want only some items to be selectable.

**Comment by yotamberk** [comment]
Eh... you're right...
I almost feel that applying this through `selectable` in the item itself is more advisable.
I want to think about this another day before I accept this PR.

**Comment by yotamberk** [comment]
You're Right! my mistake... I'm reviewing this now.

File: lib/timeline/component/item/Item.js [hunk 1]
```diff
@@ -83,6 +90,21 @@ class Item {
...
   }
 
+  /**
+   * Set whether the item can be selected.
+   * Can only be set/unset if the timeline's `selectable` configuration option is `true`.
+   * @param {Object} data `data` from `constructor` and `setData`
+   */
+  setSelectability(data) {
+    if (data) {
+      if (data.selectable !== undefined) {
```
Comment (yotamberk) [inline lib/timeline/component/item/Item.js]: `data.selectable !== undefined` will also allow `null` to  act as true.
Maybe doing something like this is better:
``` javascript
 this.selectable = typeof data.selectable === 'undefined' ? true : Boolean(data.selectable)
```

File: lib/timeline/component/item/Item.js [hunk 2]
```diff
@@ -27,6 +27,7 @@ class Item {
     this.selected = false;
     this.displayed = false;
     this.groupShowing = true;
+    this.selectable = false;
```
Comment (yotamberk) [inline lib/timeline/component/item/Item.js]: Shouldn't this be defaulted by the `options.selectable`?

Comment (jczacharia) [reply lib/timeline/component/item/Item.js]: Yes but I think it won't make a difference. I'll add it.

File: lib/timeline/component/item/Item.js (hunk 1 above)
Comment (jczacharia) [reply lib/timeline/component/item/Item.js]: Will do.

**Comment by mojoaxel** [comment]
> I've noticed a bug in the example

I don't think anymore this has anything to do with this feature. The feature seams to work like expected...

File: lib/timeline/component/item/Item.js [hunk 3]
```diff
@@ -27,6 +27,8 @@ class Item {
     this.selected = false;
     this.displayed = false;
     this.groupShowing = true;
+    console.log(options);
```
Comment (mojoaxel) [inline lib/timeline/component/item/Item.js]: see https://github.com/visjs/vis-timeline/pull/92

**Pull request merged**